src/ray/common/task/task_spec.cc

```diff
@@ -99,7 +99,8 @@ void TaskSpecification::ComputeResources() {
     // A static nil object is used here to avoid allocating the empty object every time.
     required_resources_ = ResourceSet::Nil();
   } else {
-    required_resources_.reset(new ResourceSet(MapFromProtobuf(required_resources)));
+    required_resources_ =
```

Comment:
This PR is huge, prefer to put those unrelated refactors to a separate PR to make the review easier